src/k8s/pkg/k8sd/setup/util_kubeconfig.go

```diff
@@ -38,3 +39,14 @@ func Kubeconfig(path string, token string, url string, caPEM string) error {
 	defer file.Close()
 	return renderKubeconfig(file, token, url, caPEM)
 }
+
+// ClientKubeconfig provides a kubeconfig suitable for client manipulations.
+func ClientKubeconfig(token string, url string, caPEM string) ([]byte, error) {
```

Comment:
The name `ClientKubeconfig` here does not describe the usage, and does not help distinguish from the simple `Kubeconfig` just above. Should we call this something like `KubeconfigString`? and return a `(string, error)`? 
```suggestion
// ClientKubeconfig provides a kubeconfig suitable for client manipulations.
func ClientKubeconfig(token string, url string, caPEM string) ([]byte, error) {
```